src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -1120,4 +1120,57 @@ public static boolean validateExpectedSortOrder(
         return isValid;
     }
 
+
+    /**
+     * Sets a collection of GATKReads as supplemental reads of the primary read,
+     * This tool will set the isSupplemental attribute as well as the 'SA:' tag
+     * properly according to the samspec (example: SA:primaryreadname,position,strand,Cigar,mapQ,NM;...)
+     *
+     * @param primaryRead the primary read in the set of supplemental reads
+     * @param supplementalReads an arbitrarily sorted list of reads to be set as
+     *                          supplemental to eachother and the primary read
+     *
+     * @throws UserException 
+     */
+    public static void setReadsAsSupplemental(GATKRead primaryRead, List<GATKRead> supplementalReads) {
```

Comment:
For all of the changes to this class, many of them have been addressed by my SATagBuilder class